src/enclave-agent/src/main.rs

```diff
@@ -0,0 +1,159 @@
+use std::env;
+use std::path::Path;
+use std::sync::Arc;
+
+use clap::{crate_authors, crate_version, App, Arg};
+use protocols::{image, image_ttrpc};
+use ttrpc::asynchronous::Server;
+use ttrpc::{self, error::get_rpc_status as ttrpc_error};
+
+use async_trait::async_trait;
+use tokio::signal::unix::{signal, SignalKind};
+use tokio::sync::Mutex;
+
+use anyhow::{anyhow, Result};
+use image_rs::config::ImageConfig;
+use image_rs::image::ImageClient;
+use image_rs::snapshots;
+
+const CONTAINER_BASE: &str = "/run/enclave-cc-containers";
```

Comment:
- This is quite a long path.
- If you split this into three components, we have more flexibility to  store additional state at some later stage. Something like `/run/enclave-cc/containers/` or maybe `/run/enclave-cc/ctrs/`.

Reply:
Get it. Thanks.